pkg/agent/secondarynetwork/podwatch/sriov.go

```diff
@@ -147,13 +147,16 @@ func (pc *PodController) releaseSriovVFDeviceID(podName, podNamespace, interface
 	}
 }
 
-func (pc *PodController) assignUnusedSriovVFDeviceID(podName, podNamespace, resourceName, interfaceName string) (string, error) {
+func (pc *PodController) getUnusedOrAssignedSriovVFDeviceID(podName, podNamespace, resourceName, interfaceName string) (string, error) {
 	var cache []podSriovVFDeviceIDInfo
 	cache, err := pc.buildVFDeviceIDListPerPod(podName, podNamespace)
 	if err != nil {
 		return "", err
 	}
 	for idx := range cache {
+		if cache[idx].resourceName == resourceName && cache[idx].ifName == interfaceName {
+			return cache[idx].vfDeviceID, nil
+		}
 		if cache[idx].resourceName == resourceName && cache[idx].ifName == "" {
 			// Unused PCI address found. Associate PCI address to the interface.
```

Comment:
Is it possible an entry later can match `interfaceName`? If so, we should not change the entry and return here, but should store the entry, and continue searching for `interfaceName`, and only when no matched entry we return an unused one?

Reply:
Good catch, refine the logic to find the first unused and assign it only when there is a match.